### ISSUE TITLE
Please update glog

### ISSUE THREAD

**Issue by vlazdra** [facebook/react-native#20780]
- [x] Review the documentation: https://facebook.github.io/react-native
  - [x] Search for existing issues: https://github.com/facebook/react-native/issues
  - [x] Use the latest React Native release: https://github.com/facebook/react-native/releases

## Environment:
* React Native Environment Info:
    * System:
      * OS: macOS High Sierra 10.13.6
      * CPU: x64 Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
      * Memory: 713.88 MB / 16.00 GB
      * Shell: 3.2.57 - /bin/bash
    * Binaries:
      * Node: 8.11.4 - ~/.nvm/versions/node/v8.11.4/bin/node
      * Yarn: 1.9.4 - ~/.nvm/versions/node/v8.11.4/bin/yarn
      * npm: 5.6.0 - ~/.nvm/versions/node/v8.11.4/bin/npm
    * SDKs:
      * iOS SDK:
        * Platforms: iOS 11.4, macOS 10.13, tvOS 11.4, watchOS 4.3
      * Android SDK:
        * Build Tools: 19.1.0, 21.1.2, 23.0.1, 23.0.2, 23.0.3, 24.0.1, 24.0.2, 25.0.0, 25.0.2, 25.0.3, 26.0.1, 26.0.2, 27.0.1, 27.0.2, 27.0.3, 28.0.0, 28.0.0, 28.0.2
        * API Levels: 19, 20, 21, 23, 24, 25, 26, 27, 28
    * IDEs:
      * Android Studio: 3.1 AI-173.4907809
      * Xcode: 9.4.1/9F2000 - /usr/bin/xcodebuild
    * npmPackages:
      * react: 16.4.1 => 16.4.1
      * react-native: 0.56.0 => 0.56.0
    * npmGlobalPackages:
      * babel-preset-react-native: 2.1.0
      * create-react-native-app: 1.0.0
      * react-native-cli: 2.0.1
      * react-native-create-library: 3.1.2
      * react-native: 0.56.0

## Description
I am unable to build my project with the latest version of RN and GoogleCast SDK, because of duplicate symbols declaration for glog and GoogleCast.

What I have found, and hope that would resolve the issue, is the newest version of glog: https://github.com/google/glog/releases/tag/v0.3.5
When you look at the changelog, you can see that they mentioned the collision with other of their projects.

Please update glog for RN to version 0.3.5 instead of 0.3.4.

Build output when trying to build the project:

`
duplicate symbol __ZN3fLB18FLAGS_nolog_prefixE in:
    /Users/user/Documents/Apps/test-rn/RNAwesomeProject/ios/Pods/google-cast-sdk/GoogleCastSDK-ios-4.3.1_static/GoogleCast.framework/GoogleCast(logging_f31ccd6e0091bd60840b95581a5633bf.o)
    /Users/user/Library/Developer/Xcode/DerivedData/AwesomeProject-bcowxhqhamztzfgnadpnfmbuniwk/Build/Products/Debug-iphonesimulator/glog/libglog.a(logging.o)
duplicate symbol __ZN3fLB16FLAGS_log_prefixE in:
    /Users/user/Documents/Apps/test-rn/RNAwesomeProject/ios/Pods/google-cast-sdk/GoogleCastSDK-ios-4.3.1_static/GoogleCast.framework/GoogleCast(logging_f31ccd6e0091bd60840b95581a5633bf.o)
    /Users/user/Library/Developer/Xcode/DerivedData/AwesomeProject-bcowxhqhamztzfgnadpnfmbuniwk/Build/Products/Debug-iphonesimulator/glog/libglog.a(logging.o)
duplicate symbol __ZN3fLB28FLAGS_nosymbolize_stacktraceE in:
    /Users/user/Documents/Apps/test-rn/RNAwesomeProject/ios/Pods/google-cast-sdk/GoogleCastSDK-ios-4.3.1_static/GoogleCast.framework/GoogleCast(examine_stack_04a326263e981a36fcf91c06d43ac91b.o)
    /Users/user/Library/Developer/Xcode/DerivedData/AwesomeProject-bcowxhqhamztzfgnadpnfmbuniwk/Build/Products/Debug-iphonesimulator/glog/libglog.a(utilities.o)
duplicate symbol __ZN3fLB26FLAGS_symbolize_stacktraceE in:
    /Users/user/Documents/Apps/test-rn/RNAwesomeProject/ios/Pods/google-cast-sdk/GoogleCastSDK-ios-4.3.1_static/GoogleCast.framework/GoogleCast(examine_stack_04a326263e981a36fcf91c06d43ac91b.o)
    /Users/user/Library/Developer/Xcode/DerivedData/AwesomeProject-bcowxhqhamztzfgnadpnfmbuniwk/Build/Products/Debug-iphonesimulator/glog/libglog.a(utilities.o)
duplicate symbol __ZN3fLI9FLAGS_novE in:
    /Users/user/Library/Developer/Xcode/DerivedData/AwesomeProject-bcowxhqhamztzfgnadpnfmbuniwk/Build/Products/Debug-iphonesimulator/glog/libglog.a(vlog_is_on.o)
    /Users/user/Documents/Apps/test-rn/RNAwesomeProject/ios/Pods/google-cast-sdk/GoogleCastSDK-ios-4.3.1_static/GoogleCast.framework/GoogleCast(vlog_is_on_d296ce482b5042b2314dc169f2626462.o)
duplicate symbol __ZN3fLI7FLAGS_vE in:
    /Users/user/Library/Developer/Xcode/DerivedData/AwesomeProject-bcowxhqhamztzfgnadpnfmbuniwk/Build/Products/Debug-iphonesimulator/glog/libglog.a(vlog_is_on.o)
    /Users/user/Documents/Apps/test-rn/RNAwesomeProject/ios/Pods/google-cast-sdk/GoogleCastSDK-ios-4.3.1_static/GoogleCast.framework/GoogleCast(vlog_is_on_d296ce482b5042b2314dc169f2626462.o)
duplicate symbol __ZN3fLS13FLAGS_vmoduleE in:
    /Users/user/Library/Developer/Xcode/DerivedData/AwesomeProject-bcowxhqhamztzfgnadpnfmbuniwk/Build/Products/Debug-iphonesimulator/glog/libglog.a(vlog_is_on.o)
    /Users/user/Documents/Apps/test-rn/RNAwesomeProject/ios/Pods/google-cast-sdk/GoogleCastSDK-ios-4.3.1_static/GoogleCast.framework/GoogleCast(vlog_is_on_d296ce482b5042b2314dc169f2626462.o)
ld: 7 duplicate symbols for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
`

## Reproducible Demo

https://github.com/vlazdra/RNAwesomeProject

Because GoogleCast SDK exceeded the limit for file size for GitHub, please do a `pod install` to get all the Pods to successfully reproduce the issue.

**Comment by hramos** [comment]
glog was upgraded to 0.3.5 almost two months ago in https://github.com/facebook/react-native/commit/b5fca806059e628edb504cb1bacf62e89ee6f102. Please give the 0.57 release candidate a try: https://github.com/facebook/react-native/releases/tag/v0.57.0-rc.0

**Comment by vlazdra** [comment]
Sorry but that's for Android.
For iOS its still the same version: https://github.com/facebook/react-native/blob/v0.57.0-rc.0/third-party-podspecs/glog.podspec

**Comment by hramos** [comment]
I see. Can you send a PR?

**Comment by vlazdra** [comment]
While the original issue for this bug was regarding the glog and google-cast-sdk, with further investigation I have come to the conclusion that upgrading glog to 0.3.5 wouldn't affect the issue.

The fix is to downgrade to google-cast-sdk 4.3.0.
I have filed a separate issue with the GoogleCast team hoping to get it resolved: https://issuetracker.google.com/issues/113069508

But while I was at it, I made a pull request to update glog for iOS either way, since it was already done for Android.
